### PR TITLE
Feature/nap apis

### PR DESCRIPTION
**Pull request by behroozbc**
https://github.com/slatex/sTeX-React/issues/165

### CODE REVIEW

File: packages/alea-frontend/pages/api/nap/create-answer.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import {
+  checkIfPostOrSetError,
+  executeAndEndSet500OnError,
+  getUserIdOrSetError,
+} from '../comment-utils';
+import { CreateAnswer } from '@stex-react/api';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfPostOrSetError(req, res)) return;
+  const userId = await getUserIdOrSetError(req, res);
```
Comment (abchugh) [inline packages/alea-frontend/pages/api/nap/create-answer.ts:11]: Add `if (!userId) return;`. Here and in other APIs

File: packages/alea-frontend/pages/api/nap/edit-answer.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  checkIfPostOrSetError,
+  executeAndEndSet500OnError,
+  getUserIdOrSetError,
+} from '../comment-utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfPostOrSetError(req, res)) return;
+  const userId = await getUserIdOrSetError(req, res);
+  const { answer, id } = req.body as UpdateAnswer;
+  await executeAndEndSet500OnError(
```
Comment (abchugh) [inline packages/alea-frontend/pages/api/nap/edit-answer.ts:13]: We are going to limit editing of answers for only some time, right?
Let's add a todo here

File: packages/alea-frontend/pages/api/nap/create-review-request.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { CreateReviewRequest } from '@stex-react/api';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfPostOrSetError(req, res)) return;
+  const userId = await getUserIdOrSetError(req, res);
+  const { reviewType, answerId } = req.body as CreateReviewRequest;
+  if (!reviewType || !answerId) return res.status(422).end();
+
+  await executeAndEndSet500OnError(
+    `INSERT INTO ReviewRequest (reviewType, answerId, userId) VALUES (?,?,?)`,
```
Comment (abchugh) [inline packages/alea-frontend/pages/api/nap/create-review-request.ts:16]: Consider adding `questionId` here. Would be useful to have this information here.

File: packages/api/src/lib/nap.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+export interface UpdateAnswer {
+  answer: string;
+  id: number;
+}
+export interface CreateReviewRequest {
+  answerId: number;
+  reviewType: ReviewType;
+}
+export enum ReviewType {
+  PEER = 'Peer',
```
Comment (abchugh) [inline packages/api/src/lib/nap.ts:34]: Should we add 'SELF' here?

File: packages/alea-frontend/pages/api/nap/get-student-answer.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { checkIfGetOrSetError, executeAndEndSet500OnError } from '../comment-utils';
+import { Answer, AnswerClass, Grade } from '@stex-react/api';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!checkIfGetOrSetError(req, res)) return;
+  const id = req.query.id as string;
```
Comment (abchugh) [inline packages/alea-frontend/pages/api/nap/get-student-answer.ts:7]: Please add a todo here to do proper access control. Only the student themself and the instructors should be able to view the answers. 
Peers should be able to see the answers if there is a 'peer review request' but in that case we should share the student id.

File: packages/alea-frontend/pages/api/nap/get-student-answer.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    await executeAndEndSet500OnError<Answer[]>(`select * from Answer where id=?`, [id], res)
+  )[0];
+  const grades = await executeAndEndSet500OnError<Grade[]>(
+    `select * from Grading where answerId=?`,
+    [id],
+    res
+  );
+  const gradesAnswerClasses = await executeAndEndSet500OnError<AnswerClass[]>(
+    'select * from GradingAnswerClass where gradingId in (?)',
+    [grades.flatMap((c) => c.id)],
```
Comment (abchugh) [inline packages/alea-frontend/pages/api/nap/get-student-answer.ts:18]: c.id is a string. So why is flatMap needed?

File: packages/alea-frontend/pages/api/nap/create-review-request.ts (hunk 3 above)
Comment (behroozbc) [reply packages/alea-frontend/pages/api/nap/create-review-request.ts:17]: we discussed this on the meeting.

File: packages/api/src/lib/nap.ts (hunk 4 above)
Comment (behroozbc) [reply packages/api/src/lib/nap.ts:34]: we discussed this on the meeting.

**Pull request merged**